### PR TITLE
basketball_ptr

### PR DESCRIPTION
**Pull request by SoroushBeigi**
basketball_ptr project was migrated to the last flutter and dart sdk and deprecated codes were removed
Please review the TODO I wrote

### CODE REVIEW

File: vignettes/basketball_ptr/lib/game_score.dart [hunk 1]
```diff
@@ -37,7 +39,12 @@ class _GameScoreState extends State<GameScore> with SingleTickerProviderStateMix
     _controller.addStatusListener((status) {
       if (status == AnimationStatus.completed) {
         setState(() {
-          _data = _newData;
+          //TODO: I added this condition because _data's null safety was never checked 
+          //so I concluded that it must not be null. change if needed
+          if (_newData != null) {
+            _data = _newData!;
+          }
```
Comment (esDotDev) [inline vignettes/basketball_ptr/lib/game_score.dart:46]: Seems good to me! Just remove the comment? Might as well move the `_newData = null;` inside the `if` too?

File: vignettes/basketball_ptr/lib/game_score.dart [hunk 2]
```diff
@@ -80,7 +87,12 @@ class _GameScoreState extends State<GameScore> with SingleTickerProviderStateMix
           ),
         ),
         Padding(padding: EdgeInsets.all(8)),
-        Text('-', style: TextStyle(fontSize: 28, color: ThemeInfo.accent1, fontFamily: 'FjallaOne', package: App.pkg)),
+        Text('-',
+            style: TextStyle(
+                fontSize: 28,
+                color: ThemeInfo.accent1,
+                fontFamily: 'FjallaOne',
+                package: App.pkg)),
```
Comment (esDotDev) [inline vignettes/basketball_ptr/lib/game_score.dart:95]: Make sure line length is 120 and reformat?

**Review by esDotDev — CHANGES_REQUESTED** [review]
Looks good! Just some small stuff

**Comment by SoroushBeigi** [comment]
all done!

**Review by esDotDev — APPROVED** [review]

**Pull request merged**